### PR TITLE
refactor(listbox-option): migrate to fast

### PR DESCRIPTION
**Pull request by rinaok**

### CODE REVIEW

File: libs/components/src/lib/list-item/list-item.template.ts [hunk 1]
```diff
@@ -27,17 +24,20 @@ export const ListItemTemplate: (
...
 		aria-setsize="${x => x.ariaSetSize}"
 		role="option">
-		${x => affixIconTemplate(x.icon)}
+		<slot name="start">${x => affixIconTemplate(x.icon)}</slot>
 		<div class="text-content">
 			${when(x => x.textPrimary, html`<div class="text-primary">${x => x.textPrimary}</div>`)}
 			${when(x => x.textSecondary, html`<div class="text-secondary">${x => x.textSecondary}</div>`)}
 		</div>
-		<slot name="meta"></slot>
+		<slot name="end"><slot name="meta"></slot></slot>
```
Comment (yinonov) [inline libs/components/src/lib/list-item/list-item.template.ts:41]: 2 slots?

File: libs/components/src/lib/list-item/list-item.template.ts [hunk 2]
```diff
@@ -27,17 +24,20 @@ export const ListItemTemplate: (
...
+		?checked="${(x) => x.checked}"
+		?disabled="${(x) => x.disabled}"
 		aria-checked="${x => x.ariaChecked}"
 		aria-disabled="${x => x.ariaDisabled}"
 		aria-posinset="${x => x.ariaPosInSet}"
 		aria-selected="${x => x.ariaSelected}"
 		aria-setsize="${x => x.ariaSetSize}"
 		role="option">
-		${x => affixIconTemplate(x.icon)}
+		<slot name="start">${x => affixIconTemplate(x.icon)}</slot>
```
[the remark this answers is not in the document]
Comment (rinaok) [reply libs/components/src/lib/list-item/list-item.template.ts]: @yinonov We don't have to use them.
It's from fasts listbox-option api:

Screen Shot 2022-08-17 at 12 21 19

File: libs/components/src/lib/list-item/list-item.template.ts (hunk 1 above)
Comment (rinaok) [reply libs/components/src/lib/list-item/list-item.template.ts]: @yinonov I wanted to use fast's slots from their API but didn't want to use the name "end" for our api.
Will it be better to leave just our slots?

image

**Comment by rinaok** [comment]
@yinonov @jshenkman @AyalaBu 

This is how the button inside a list-item looks in vivid-2, how should it look in vivid-3?

image

File: libs/components/src/lib/listbox-option/listbox-option.template.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	return html`
+	<div class="${getClasses}">
+		<div class="option-wrapper"
+			?selected="${(x) => x.selected}"
+			?disabled="${(x) => x.disabled}"
+			aria-disabled="${x => x.ariaDisabled}"
+			aria-posinset="${x => x.ariaPosInSet}"
+			aria-selected="${x => x.ariaSelected}"
+			aria-setsize="${x => x.ariaSetSize}"
+			role="option">
```
Comment (rachelbt) [inline libs/components/src/lib/listbox-option/listbox-option.template.ts:37]: maybe all these one needs to move to the upper div as the css classes moved

File: libs/components/src/lib/listbox-option/listbox-option.scss [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	hyphens: auto;
+	min-block-size: $min-block-size;
+	padding-block: 10px;
+	padding-inline: $gap;
+	text-decoration: none;
+	vertical-align: middle;
+	word-break: break-word;
+
+	.option-wrapper {
+		display: inline-flex;
```
Comment (rachelbt) [inline libs/components/src/lib/listbox-option/listbox-option.scss:39]:
```suggestion
		display: flex;
```

File: libs/components/src/lib/listbox-option/listbox-option.template.ts (hunk 3 above)
Comment (rinaok) [reply libs/components/src/lib/listbox-option/listbox-option.template.ts:37]: Yes I'll move it

File: libs/components/src/lib/listbox-option/index.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import '../icon';
+
+import type { FoundationElementDefinition } from '@microsoft/fast-foundation';
+import { designSystem } from '../../shared/design-system';
+import { ListboxOption } from './listbox-option';
+import { ListboxOptionTemplate as template } from './listbox-option.template';
+import styles from './listbox-option.scss';
+
+export const vividListboxOption = ListboxOption.compose<FoundationElementDefinition>({
+	baseName: 'listbox-option',
```
Comment (yinonov) [inline libs/components/src/lib/listbox-option/index.ts:11]: should simply be named option as is also used within select component
```suggestion
	baseName: 'option',
```

**Review by yinonov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**